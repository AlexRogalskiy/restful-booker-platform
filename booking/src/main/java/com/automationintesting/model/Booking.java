package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Entity;
import javax.validation.Valid;
import javax.validation.constraints.*;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Date;

@Entity
public class Booking {

    @JsonProperty
    private int bookingid;

    @JsonProperty
    private int roomid;

    @JsonProperty
    @NotBlank(message = "Firstname should not be blank")
    @NotNull(message = "Firstname should not be null")
    @Size(min = 3, max = 18)
    private String firstname;

    @JsonProperty
    @NotBlank(message = "Lastname should not be blank")
    @NotNull(message = "Lastname should not be null")
    @Size(min = 3, max = 30)
    private String lastname;

    @JsonProperty
    @NotNull(message = "Total price should not be null")
    @Min(10)
    @Max(999)
    private int totalprice;

    @JsonProperty
    @NotNull(message = "Deposit paid should not be null")
    private boolean depositpaid;

    @JsonProperty(value = "bookingdates")
    @Valid
    private BookingDates bookingDates;

    public Booking(int bookingid, int roomid, String firstname, String lastname, int totalprice, boolean depositpaid, BookingDates bookingDates) {
        this.bookingid = bookingid;
        this.roomid = roomid;
        this.firstname = firstname;
        this.lastname = lastname;
        this.totalprice = totalprice;
        this.depositpaid = depositpaid;
        this.bookingDates = bookingDates;
    }

    public Booking(ResultSet result) throws SQLException {
        this.bookingid = result.getInt("bookingid");
        this.roomid = result.getInt("roomid");
        this.firstname = result.getString("firstname");
        this.lastname = result.getString("lastname");
        this.totalprice = result.getInt("totalprice");
        this.depositpaid = result.getBoolean("depositpaid");
        this.bookingDates = new BookingDates(result.getDate("checkin"), result.getDate("checkout"));
    }

    public Booking() {
    }

    public String getFirstname() {
        return firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public int getTotalprice() {
        return totalprice;
    }

    public boolean isDepositpaid() {
        return depositpaid;
    }

    public BookingDates getBookingDates() {
        return bookingDates;
    }

    public int getRoomid() {
        return roomid;
    }

    public int getBookingid() {
        return bookingid;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public void setTotalprice(int totalprice) {
        this.totalprice = totalprice;
    }

    public void setDepositpaid(boolean depositpaid) {
        this.depositpaid = depositpaid;
    }

    public void setBookingDates(BookingDates bookingDates) {
        this.bookingDates = bookingDates;
    }

    public void setRoomid(int roomid) {
        this.roomid = roomid;
    }

    public void setBookingid(int bookingid) {
        this.bookingid = bookingid;
    }

    @Override
    public String toString() {
        return "Booking{" +
                "roomid=" + roomid +
                ", firstname='" + firstname + '\'' +
                ", lastname='" + lastname + '\'' +
                ", totalprice=" + totalprice +
                ", depositpaid=" + depositpaid +
                ", bookingDates=" + bookingDates +
                '}';
    }

    public static class BookingBuilder {

        private int roomid;
        private String firstname;
        private String lastname;
        private int totalprice;
        private boolean depositpaid;
        private Date checkin;
        private Date checkout;

        public BookingBuilder setRoomid(int roomid){
            this.roomid = roomid;

            return this;
        }

        public BookingBuilder setFirstname(String firstname) {
            this.firstname = firstname;

            return this;
        }

        public BookingBuilder setLastname(String lastname) {
            this.lastname = lastname;

            return this;
        }

        public BookingBuilder setTotalprice(int totalprice) {
            this.totalprice = totalprice;

            return this;
        }

        public BookingBuilder setDepositpaid(boolean depositpaid) {
            this.depositpaid = depositpaid;

            return this;
        }

        public BookingBuilder setCheckin(Date checkin) {
            this.checkin = checkin;

            return this;
        }

        public BookingBuilder setCheckout(Date checkout) {
            this.checkout = checkout;

            return this;
        }

        public Booking build(){
            BookingDates bookingDates = new BookingDates(checkin, checkout);

            return new Booking(0, roomid, firstname, lastname, totalprice, depositpaid, bookingDates);
        }
    }
}
