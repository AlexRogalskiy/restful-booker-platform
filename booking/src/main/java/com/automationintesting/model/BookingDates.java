package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Entity;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity
public class BookingDates {

    @JsonProperty
    @NotBlank(message = "Checkin should not be blank")
    @NotNull(message = "Checkin should not be null")
    private Date checkin;

    @JsonProperty
    @NotBlank(message = "Checkout should not be blank")
    @NotNull(message = "Checkout should not be null")
    private Date checkout;

    public BookingDates() {
    }

    public BookingDates(Date checkin, Date checkout) {
        this.checkin = checkin;
        this.checkout = checkout;
    }

    public Date getCheckin() {
        return checkin;
    }

    public void setCheckin(Date checkin) {
        this.checkin = checkin;
    }

    public Date getCheckout() {
        return checkout;
    }

    public void setCheckout(Date checkout) {
        this.checkout = checkout;
    }

    @Override
    public String toString() {
        return "BookingDates{" +
                "checkin=" + checkin +
                ", checkout=" + checkout +
                '}';
    }
}
