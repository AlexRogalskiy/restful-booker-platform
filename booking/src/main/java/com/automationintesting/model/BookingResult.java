package com.automationintesting.model;

import org.springframework.http.HttpStatus;

public class BookingResult {

    private Booking booking;

    private HttpStatus result;

    public BookingResult(Booking booking, HttpStatus result) {
        this.booking = booking;
        this.result = result;
    }

    public HttpStatus getStatus() {
        return result;
    }

    public Booking getBooking() {
        return booking;
    }
}
