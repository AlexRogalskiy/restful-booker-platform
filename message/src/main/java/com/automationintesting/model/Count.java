package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Count {

    @JsonProperty
    private int count;

    public Count(int count) {
        this.count = count;
    }

    public int getCount() {
        return count;
    }
}
