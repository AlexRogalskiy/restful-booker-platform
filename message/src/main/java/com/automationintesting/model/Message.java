package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.ResultSet;
import java.sql.SQLException;

public class Message {

    @JsonProperty
    private int messageid;

    @JsonProperty
    private String name;

    @JsonProperty
    private String email;

    @JsonProperty
    private String phone;

    @JsonProperty
    private String subject;

    @JsonProperty
    private String description;

    public Message() {
    }

    public Message(String name, String email, String phone, String subject, String description) {
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.subject = subject;
        this.description = description;
    }

    public Message(ResultSet result) throws SQLException {
        this.messageid = result.getInt("messageid");
        this.name = result.getString("name");
        this.email = result.getString("email");
        this.phone = result.getString("phone");
        this.subject = result.getString("subject");
        this.description = result.getString("description");
    }

    public int getMessageid() {
        return messageid;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getPhone() {
        return phone;
    }

    public String getSubject() {
        return subject;
    }

    public String getDescription() {
        return description;
    }

    public void setMessageid(int messageid) {
        this.messageid = messageid;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Message{" +
                "messageid=" + messageid +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", phone='" + phone + '\'' +
                ", subject='" + subject + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}
