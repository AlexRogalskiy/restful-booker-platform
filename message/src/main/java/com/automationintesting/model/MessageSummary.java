package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.ResultSet;
import java.sql.SQLException;

public class MessageSummary {

    @JsonProperty
    private int id;

    @JsonProperty
    private String name;

    @JsonProperty
    private String subject;

    public MessageSummary(ResultSet resultSet) throws SQLException {
        this.id = resultSet.getInt("messageid");
        this.name = resultSet.getString("name");
        this.subject = resultSet.getString("subject");
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getSubject() {
        return subject;
    }

    @Override
    public String toString() {
        return "MessageSummary{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", subject='" + subject + '\'' +
                '}';
    }
}
