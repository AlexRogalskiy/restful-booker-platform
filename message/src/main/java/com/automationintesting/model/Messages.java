package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.List;

public class Messages {

    @JsonProperty
    private List<MessageSummary> messages;

    public Messages(List<MessageSummary> messages) {
        this.messages = messages;
    }

    public List<MessageSummary> getMessages() {
        return messages;
    }

}
