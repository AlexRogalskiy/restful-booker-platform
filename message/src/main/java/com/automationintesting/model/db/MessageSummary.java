package com.automationintesting.model.db;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.sql.ResultSet;
import java.sql.SQLException;

public class MessageSummary {

    @JsonProperty
    private int id;

    @JsonProperty
    private String name;

    @JsonProperty
    private String subject;

    public MessageSummary(ResultSet resultSet) throws SQLException {
        this.id = resultSet.getInt("messageid");
        this.name = resultSet.getString("name");
        this.subject = resultSet.getString("subject");
    }

    public MessageSummary(int id, String name, String subject) {
        this.id = id;
        this.name = name;
        this.subject = subject;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getSubject() {
        return subject;
    }

    @Override
    public String toString() {
        return "MessageSummary{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", subject='" + subject + '\'' +
                '}';
    }
}
