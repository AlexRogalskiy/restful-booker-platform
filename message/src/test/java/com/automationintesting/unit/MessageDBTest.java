package com.automationintesting.unit;

import com.automationintesting.model.Message;
import com.automationintesting.model.MessageSummary;
import liquibase.exception.LiquibaseException;
import org.junit.Before;
import org.junit.Test;

import java.sql.SQLException;
import java.util.List;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.junit.Assert.assertEquals;

public class MessageDBTest extends BaseTest {

    private int currentMessageId;

    @Before
    public void resetDB() throws SQLException, LiquibaseException {
        messageDB.resetDB();

        Message message = new Message("James",
                "james@dean.com",
                "01821 123321",
                "Just getting a message setup",
                "Lorem ipsum dolores est");

        Message createdMessage = messageDB.create(message);

        currentMessageId = createdMessage.getMessageid();
    }

    @Test
    public void testQueryMessage() throws SQLException {
        Message queriedMessage = messageDB.query(currentMessageId);

        String queriedMessageString = queriedMessage.toString();

        assertEquals("Message{messageid=" + currentMessageId + ", name='James', email='james@dean.com', phone='01821 123321', subject='Just getting a message setup', description='Lorem ipsum dolores est'}", queriedMessageString);
    }

    @Test
    public void testCreateMessage() throws SQLException {
        Message message = new Message("Mark",
                "mark@mwtestconsultancy.co.uk",
                "01821 912812",
                "A subject you may be interested in",
                "In posuere accumsan aliquet.");

        Message createdMessage = messageDB.create(message);

        String createdMessageString = createdMessage.toString();

        assertEquals("Message{messageid=" + (currentMessageId  + 1) + ", name='Mark', email='mark@mwtestconsultancy.co.uk', phone='01821 912812', subject='A subject you may be interested in', description='In posuere accumsan aliquet.'}", createdMessageString);
    }

    @Test
    public void testDeleteMessage() throws SQLException {
        boolean isDeleted = messageDB.delete(currentMessageId);

        assertThat(isDeleted, is(true));
    }

    @Test
    public void testGetMessages() throws SQLException {
        Message message = new Message("Mark",
                "mark@mwtestconsultancy.co.uk",
                "01821 912812",
                "A subject you may be interested in",
                "In posuere accumsan aliquet.");

        messageDB.create(message);

        List<MessageSummary> messageSummaries = messageDB.queryMessages();

        assertEquals("[MessageSummary{id=1, name='James Dean', subject='Booking enquiry'}, MessageSummary{id=2, name='James', subject='Just getting a message setup'}, MessageSummary{id=3, name='Mark', subject='A subject you may be interested in'}]", messageSummaries.toString());
    }

    @Test
    public void testReadCount() throws SQLException {
        int currentMessageCount = messageDB.getUnreadCount();

        assertThat(currentMessageCount, is(2));
    }

    @Test
    public void testMarkAsRead() throws SQLException {
        int beforeReadCount = messageDB.getUnreadCount();

        messageDB.markAsRead(currentMessageId);

        int afterReadCount = messageDB.getUnreadCount();

        assertThat(afterReadCount, is(beforeReadCount - 1));
    }

}
