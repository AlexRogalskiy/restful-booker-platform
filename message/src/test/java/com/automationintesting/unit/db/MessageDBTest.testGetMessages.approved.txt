[MessageSummary{id=1, name='James Dean', subject='Booking enquiry'}, MessageSummary{id=2, name='James', subject='Just getting a message setup'}, MessageSummary{id=3, name='Mark', subject='A subject you may be interested in'}]
